### ISSUE TITLE
require.onError is called for the first two missing modules ... but silent fail for any missing modules after

### ISSUE THREAD

**Issue by lemonskip** [requirejs/requirejs#871]
Hi,

My title basically says it all but I will try to explain my specific context ... but i believe it to be a general problem:

So My application setup is as follows (two separate code bases):
- Main app (comes with requirejs and third party deps that will be used by a plugin)
- A plugin (has third party deps but is not included in the build)

Disclaimer: Please excuse any syntax errors ... thats not the issue. Code I am working on is private, however the issue should be beneficial for all.

When the plugin is loaded by the main app, I create a multi context akin to the following example:

```
var pluginContextId = '*uniquePluginId*';
var pluginRequire = require.config( amd_config = {
                context: pluginContextId,
                baseUrl: '/javascript/*pluginLocation*',
                setTimeout: 100000000000000000 // should not need to do this ... but during debugging I felt the need to set this so
            };

// cache it for later
cachePluginContext( pluginContextId, pluginRequire );

// load the plugin
pluginRequire([ 'main' ], function() {} );

```

Using some not so ideal logic I can now register a global onError callback to handle missing thrid-party dependencies of the plugin (because by default they will fail!):

```
/**
 * Global Error Handling. Searches for failedIds of plugins 
 * and tries to load it from console. Failing that ...
 * You will get #scripterror
 */
require.onError = function( err ){

    if( err
        && err.requireModules 
        && err.requireModules.length > 0 ) {
        var failedId = err.requireModules && err.requireModules[0];

        // use the script attribute 'data-requirecontext' to locate a reference to the 
        // plugin require context #IKnowNotClean
        var pluginRequire = getPluginRequire( err );

        pluginRequire.undef(failedId);

        // load missing dep from Main App
        require([ failedId ], function ( Module ) {
            // Now redefine within the context of the plugin
            define( failedId, [], function(){ return Module; } );
            pluginRequire([failedId], function () {} );

        }, function( err ) {
            // OH well i guess it really does not exist!
            console.error( "Error: Main App does not have plugin dependency", err );
        });
    }

}
```

This is similar to the errback example provided in the requirejs api:
http://requirejs.org/docs/api.html#errbacks

And it surprisingly it works very nicely (albeit would be lovely to avoid http requests when module built ... #MayBeIWillLookIntoAlmond ... #IDigress).

So Whats the issue?

I if the missing third-party deps of the plugin is 2 or less, then this solution is good. If it is more ... it timesout! Why? #IDontKnow

Solution I'm going with for now:
So after extensively using debugger; all over the place and getting stuck/lost when the stack trace kept referring to anonymous bind definitions :( I modified the following in the require.js code (line 516 v2.1.8):

```
        function onError(err, errback) {
            var ids = err.requireModules,
                notified = false;

            if (errback) {
                errback(err);
            } else {
                each(ids, function (id) {
                    var mod = getOwn(registry, id);
                    if (mod) {
                        //Set error on module, so it skips timeout checks.
                        mod.error = err;
                        if (mod.events.error) {
                            notified = true;
                            // !Hack: This is a Hack!
                            // mod.emit('error', err);
                        }
                    }
                });

                // !Hack: This is a Hack!
                if (true || !notified) {
                    req.onError(err);
                }
            }
        }
```

In short I the intension here is to force all errors to call the global onError function.

I am positing this as an issue, because:
1. It should work as expected without my hack
2. Its not elegant
3. Error handling in requirejs is seemingly limited 

I'd appreciate your comment/thoughts on this if you have a moment.

Thanks in advance

M.
